### ISSUE TITLE
when I type `d`，something went wrong.

### ISSUE THREAD

**Issue by JYInMyHeart** [PowerShell/PSReadLine#1944]
Oops, something went wrong.
Please report this bug with ALL the details below, including both the 'Environment' and 'Exception' sections.
Please report on GitHub: https://github.com/PowerShell/PSReadLine/issues/new?template=Bug_Report.md
Thank you!

### Environment
PSReadLine: 2.1.0-rc1
PowerShell: 7.1.0-rc.2
OS: Microsoft Windows 10.0.19042
BufferWidth: 120
BufferHeight: 30

Last 54 Keys
```
 Enter
 Enter
 Enter
 d l s Enter
 c d Spacebar . . Enter
 d Enter
 Enter
 Enter
 Enter
 l s Enter
 UpArrow RightArrow RightArrow Spacebar / q Enter
 UpArrow RightArrow RightArrow RightArrow RightArrow RightArrow LeftArrow Backspace \ Enter
 l s Spacebar / ? Enter
 " d i r Backspace Backspace Backspace Backspace Backspace d
```

### Exception
```
System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at System.Text.StringBuilder.get_Chars(Int32 index)
   at Microsoft.PowerShell.PSConsoleReadLine.ConvertOffsetToPoint(Int32 offset)
   at Microsoft.PowerShell.PSConsoleReadLine.ReallyRender(RenderData renderData, String defaultColor)
   at Microsoft.PowerShell.PSConsoleReadLine.ForceRender()
   at Microsoft.PowerShell.PSConsoleReadLine.Insert(Char c)
   at Microsoft.PowerShell.PSConsoleReadLine.SelfInsert(Nullable`1 key, Object arg)
   at Microsoft.PowerShell.PSConsoleReadLine.ProcessOneKey(PSKeyInfo key, Dictionary`2 dispatchTable, Boolean ignoreIfNoAction, Object arg)
   at Microsoft.PowerShell.PSConsoleReadLine.InputLoop()
   at Microsoft.PowerShell.PSConsoleReadLine.ReadLine(Runspace runspace, EngineIntrinsics engineIntrinsics, CancellationToken cancellationToken)
```

**Comment by daxian-dbw** [comment]
@JYInMyHeart Thanks for reporting the issue!
I cannot reproduce the issue based on your last 54 keys. Can you please provide a repro? That would be very helpful to the investigation. Thank you in advance.

**Comment by JYInMyHeart** [comment]
Sorry. It's my mistake. I had copied lots of words to run a program. These words start with 'd'. So when I typed 'd' it would cause an error because of auto-completion. Maybe these words are really too long.

**Comment by daxian-dbw** [comment]
@JYInMyHeart No need to be sorry at all! It's obviously a bug in PSReadLine :)
Can you share the words that you copied (the copy that starts with 'd')? It will help to repro the issue in my local environment so that I can fix the issue.

**Comment by JYInMyHeart** [comment]
The words is below.


d:\Users\alex\.jdks\adopt-openjdk-1.8.0_272\bin\java.exe -javaagent:D:\jetbrains\apps\IDEA-U\ch-0\203.5600.34\lib\idea_rt.jar=55579:D:\jetbrains\apps\IDEA-U\ch-0\203.5600.34\bin -Dfile.encoding=UTF-8 -classpath D:\Users\alex\.jdks\adopt-openjdk-1.8.0_272\jre\lib\charsets.jar;D:\Users\alex\.jdks\adopt-openjdk-1.8.0_272\jre\lib\ext\access-bridge-64.jar;D:\Users\alex\.jdks\adopt-openjdk-1.8.0_272\jre\lib\ext\cldrdata.jar;D:\Users\alex\.jdks\adopt-openjdk-1.8.0_272\jre\lib\ext\dnsns.jar;D:\Users\alex\.jdks\adopt-openjdk-1.8.0_272\jre\lib\ext\jaccess.jar;D:\Users\alex\.jdks\adopt-openjdk-1.8.0_272\jre\lib\ext\localedata.jar;D:\Users\alex\.jdks\adopt-openjdk-1.8.0_272\jre\lib\ext\nashorn.jar;D:\Users\alex\.jdks\adopt-openjdk-1.8.0_272\jre\lib\ext\sunec.jar;D:\Users\alex\.jdks\adopt-openjdk-1.8.0_272\jre\lib\ext\sunjce_provider.jar;D:\Users\alex\.jdks\adopt-openjdk-1.8.0_272\jre\lib\ext\sunmscapi.jar;D:\Users\alex\.jdks\adopt-openjdk-1.8.0_272\jre\lib\ext\sunpkcs11.jar;D:\Users\alex\.jdks\adopt-openjdk-1.8.0_272\jre\lib\ext\zipfs.jar;D:\Users\alex\.jdks\adopt-openjdk-1.8.0_272\jre\lib\jce.jar;D:\Users\alex\.jdks\adopt-openjdk-1.8.0_272\jre\lib\jfr.jar;D:\Users\alex\.jdks\adopt-openjdk-1.8.0_272\jre\lib\jsse.jar;D:\Users\alex\.jdks\adopt-openjdk-1.8.0_272\jre\lib\management-agent.jar;D:\Users\alex\.jdks\adopt-openjdk-1.8.0_272\jre\lib\resources.jar;D:\Users\alex\.jdks\adopt-openjdk-1.8.0_272\jre\lib\rt.jar;D:\yoyo\code\console-client\target\classes;D:\Users\alex\.m2\repository\org\springframework\boot\spring-boot-starter\2.1.0.RELEASE\spring-boot-starter-2.1.0.RELEASE.jar;D:\Users\alex\.m2\repository\org\springframework\boot\spring-boot\2.1.0.RELEASE\spring-boot-2.1.0.RELEASE.jar;D:\Users\alex\.m2\repository\org\springframework\spring-context\5.1.2.RELEASE\spring-context-5.1.2.RELEASE.jar;D:\Users\alex\.m2\repository\org\springframework\boot\spring-boot-autoconfigure\2.1.0.RELEASE\spring-boot-autoconfigure-2.1.0.RELEASE.jar;D:\Users\alex\.m2\repository\org\springframework\boot\spring-boot-starter-logging\2.1.0.RELEASE\spring-boot-starter-logging-2.1.0.RELEASE.jar;D:\Users\alex\.m2\repository\org\apache\logging\log4j\log4j-to-slf4j\2.11.1\log4j-to-slf4j-2.11.1.jar;D:\Users\alex\.m2\repository\org\slf4j\jul-to-slf4j\1.7.25\jul-to-slf4j-1.7.25.jar;D:\Users\alex\.m2\repository\javax\annotation\javax.annotation-api\1.3.2\javax.annotation-api-1.3.2.jar;D:\Users\alex\.m2\repository\org\springframework\spring-core\5.1.2.RELEASE\spring-core-5.1.2.RELEASE.jar;D:\Users\alex\.m2\repository\org\springframework\spring-jcl\5.1.2.RELEASE\spring-jcl-5.1.2.RELEASE.jar;D:\Users\alex\.m2\repository\org\yaml\snakeyaml\1.23\snakeyaml-1.23.jar;D:\Users\alex\.m2\repository\org\springframework\boot\spring-boot-starter-web\2.1.0.RELEASE\spring-boot-starter-web-2.1.0.RELEASE.jar;D:\Users\alex\.m2\repository\org\springframework\boot\spring-boot-starter-json\2.1.0.RELEASE\spring-boot-starter-json-2.1.0.RELEASE.jar;D:\Users\alex\.m2\repository\com\fasterxml\jackson\core\jackson-databind\2.9.7\jackson-databind-2.9.7.jar;D:\Users\alex\.m2\repository\com\fasterxml\jackson\core\jackson-annotations\2.9.0\jackson-annotations-2.9.0.jar;D:\Users\alex\.m2\repository\com\fasterxml\jackson\core\jackson-core\2.9.7\jackson-core-2.9.7.jar;D:\Users\alex\.m2\repository\com\fasterxml\jackson\datatype\jackson-datatype-jdk8\2.9.7\jackson-datatype-jdk8-2.9.7.jar;D:\Users\alex\.m2\repository\com\fasterxml\jackson\datatype\jackson-datatype-jsr310\2.9.7\jackson-datatype-jsr310-2.9.7.jar;D:\Users\alex\.m2\repository\com\fasterxml\jackson\module\jackson-module-parameter-names\2.9.7\jackson-module-parameter-names-2.9.7.jar;D:\Users\alex\.m2\repository\org\hibernate\validator\hibernate-validator\6.0.13.Final\hibernate-validator-6.0.13.Final.jar;D:\Users\alex\.m2\repository\javax\validation\validation-api\2.0.1.Final\validation-api-2.0.1.Final.jar;D:\Users\alex\.m2\repository\org\jboss\logging\jboss-logging\3.3.2.Final\jboss-logging-3.3.2.Final.jar;D:\Users\alex\.m2\repository\com\fasterxml\classmate\1.4.0\classmate-1.4.0.jar;D:\Users\alex\.m2\repository\org\springframework\spring-web\5.1.2.RELEASE\spring-web-5.1.2.RELEASE.jar;D:\Users\alex\.m2\repository\org\springframework\spring-beans\5.1.2.RELEASE\spring-beans-5.1.2.RELEASE.jar;D:\Users\alex\.m2\repository\org\springframework\spring-webmvc\5.1.2.RELEASE\spring-webmvc-5.1.2.RELEASE.jar;D:\Users\alex\.m2\repository\org\springframework\spring-aop\5.1.2.RELEASE\spring-aop-5.1.2.RELEASE.jar;D:\Users\alex\.m2\repository\org\springframework\spring-expression\5.1.2.RELEASE\spring-expression-5.1.2.RELEASE.jar;D:\Users\alex\.m2\repository\org\springframework\boot\spring-boot-starter-jetty\2.1.0.RELEASE\spring-boot-starter-jetty-2.1.0.RELEASE.jar;D:\Users\alex\.m2\repository\org\eclipse\jetty\jetty-servlets\9.4.12.v20180830\jetty-servlets-9.4.12.v20180830.jar;D:\Users\alex\.m2\repository\org\eclipse\jetty\jetty-continuation\9.4.12.v20180830\jetty-continuation-9.4.12.v20180830.jar;D:\Users\alex\.m2\repository\org\eclipse\jetty\jetty-http\9.4.12.v20180830\jetty-http-9.4.12.v20180830.jar;D:\Users\alex\.m2\repository\org\eclipse\jetty\jetty-util\9.4.12.v20180830\jetty-util-9.4.12.v20180830.jar;D:\Users\alex\.m2\repository\org\eclipse\jetty\jetty-io\9.4.12.v20180830\jetty-io-9.4.12.v20180830.jar;D:\Users\alex\.m2\repository\org\eclipse\jetty\jetty-webapp\9.4.12.v20180830\jetty-webapp-9.4.12.v20180830.jar;D:\Users\alex\.m2\repository\org\eclipse\jetty\jetty-xml\9.4.12.v20180830\jetty-xml-9.4.12.v20180830.jar;D:\Users\alex\.m2\repository\org\eclipse\jetty\jetty-servlet\9.4.12.v20180830\jetty-servlet-9.4.12.v20180830.jar;D:\Users\alex\.m2\repository\org\eclipse\jetty\jetty-security\9.4.12.v20180830\jetty-security-9.4.12.v20180830.jar;D:\Users\alex\.m2\repository\org\eclipse\jetty\jetty-server\9.4.12.v20180830\jetty-server-9.4.12.v20180830.jar;D:\Users\alex\.m2\repository\org\eclipse\jetty\websocket\websocket-server\9.4.12.v20180830\websocket-server-9.4.12.v20180830.jar;D:\Users\alex\.m2\repository\org\eclipse\jetty\websocket\websocket-common\9.4.12.v20180830\websocket-common-9.4.12.v20180830.jar;D:\Users\alex\.m2\repository\org\eclipse\jetty\websocket\websocket-api\9.4.12.v20180830\websocket-api-9.4.12.v20180830.jar;D:\Users\alex\.m2\repository\org\eclipse\jetty\websocket\websocket-client\9.4.12.v20180830\websocket-client-9.4.12.v20180830.jar;D:\Users\alex\.m2\repository\org\eclipse\jetty\jetty-client\9.4.12.v20180830\jetty-client-9.4.12.v20180830.jar;D:\Users\alex\.m2\repository\org\eclipse\jetty\websocket\websocket-servlet\9.4.12.v20180830\websocket-servlet-9.4.12.v20180830.jar;D:\Users\alex\.m2\repository\javax\servlet\javax.servlet-api\4.0.1\javax.servlet-api-4.0.1.jar;D:\Users\alex\.m2\repository\org\eclipse\jetty\websocket\javax-websocket-server-impl\9.4.12.v20180830\javax-websocket-server-impl-9.4.12.v20180830.jar;D:\Users\alex\.m2\repository\org\eclipse\jetty\jetty-annotations\9.4.12.v20180830\jetty-annotations-9.4.12.v20180830.jar;D:\Users\alex\.m2\repository\org\eclipse\jetty\jetty-plus\9.4.12.v20180830\jetty-plus-9.4.12.v20180830.jar;D:\Users\alex\.m2\repository\org\ow2\asm\asm\6.2\asm-6.2.jar;D:\Users\alex\.m2\repository\org\ow2\asm\asm-commons\6.2\asm-commons-6.2.jar;D:\Users\alex\.m2\repository\org\ow2\asm\asm-tree\6.2\asm-tree-6.2.jar;D:\Users\alex\.m2\repository\org\ow2\asm\asm-analysis\6.2\asm-analysis-6.2.jar;D:\Users\alex\.m2\repository\org\eclipse\jetty\websocket\javax-websocket-client-impl\9.4.12.v20180830\javax-websocket-client-impl-9.4.12.v20180830.jar;D:\Users\alex\.m2\repository\javax\websocket\javax.websocket-api\1.1\javax.websocket-api-1.1.jar;D:\Users\alex\.m2\repository\org\mortbay\jasper\apache-el\8.5.33\apache-el-8.5.33.jar;D:\Users\alex\.m2\repository\org\springframework\boot\spring-boot-configuration-processor\2.1.0.RELEASE\spring-boot-configuration-processor-2.1.0.RELEASE.jar;D:\Users\alex\.m2\repository\org\projectlombok\lombok\1.16.20\lombok-1.16.20.jar;D:\Users\alex\.m2\repository\com\alibaba\fastjson\1.2.4\fastjson-1.2.4.jar;D:\Users\alex\.m2\repository\ch\qos\logback\logback-classic\1.2.3\logback-classic-1.2.3.jar;D:\Users\alex\.m2\repository\ch\qos\logback\logback-core\1.2.3\logback-core-1.2.3.jar;D:\Users\alex\.m2\repository\org\slf4j\slf4j-api\1.7.25\slf4j-api-1.7.25.jar;D:\Users\alex\.m2\repository\commons-io\commons-io\2.4\commons-io-2.4.jar;D:\Users\alex\.m2\repository\org\springframework\boot\spring-boot-starter-websocket\2.1.0.RELEASE\spring-boot-starter-websocket-2.1.0.RELEASE.jar;D:\Users\alex\.m2\repository\org\springframework\spring-messaging\5.1.2.RELEASE\spring-messaging-5.1.2.RELEASE.jar;D:\Users\alex\.m2\repository\org\springframework\spring-websocket\5.1.2.RELEASE\spring-websocket-5.1.2.RELEASE.jar;D:\Users\alex\.m2\repository\com\github\oshi\oshi-core\5.2.5\oshi-core-5.2.5.jar;D:\Users\alex\.m2\repository\net\java\dev\jna\jna-platform\5.6.0\jna-platform-5.6.0.jar;D:\Users\alex\.m2\repository\net\java\dev\jna\jna\5.6.0\jna-5.6.0.jar;D:\Users\alex\.m2\repository\com\yoyosys\remotecontroller\2.2.13\remotecontroller-2.2.13.jar;D:\Users\alex\.m2\repository\org\apache\commons\commons-lang3\3.8.1\commons-lang3-3.8.1.jar;D:\Users\alex\.m2\repository\commons-net\commons-net\3.6\commons-net-3.6.jar;D:\Users\alex\.m2\repository\org\jinterop\j-interop\3.0.0\j-interop-3.0.0.jar;D:\Users\alex\.m2\repository\org\jinterop\j-interopdeps\3.0.0\j-interopdeps-3.0.0.jar;D:\Users\alex\.m2\repository\org\samba\jcifs\jcifs\1.3.14-kohsuke-1\jcifs-1.3.14-kohsuke-1.jar;D:\Users\alex\.m2\repository\com\jcraft\jsch\0.1.54\jsch-0.1.54.jar;D:\Users\alex\.m2\repository\org\apache\logging\log4j\log4j-core\2.11.1\log4j-core-2.11.1.jar;D:\Users\alex\.m2\repository\org\apache\logging\log4j\log4j-api\2.11.1\log4j-api-2.11.1.jar;D:\Users\alex\.m2\repository\org\slf4j\slf4j-log4j12\1.7.25\slf4j-log4j12-1.7.25.jar;D:\Users\alex\.m2\repository\log4j\log4j\1.2.17\log4j-1.2.17.jar com.yoyosys.utils.LocalCommandUtil




I cpoid a commnad from idea's terminal.

**Comment by ghost** [comment]
:tada: This issue was addressed in 2892, which has now been successfully released in [`v2.2.0-beta4`](https://github.com/PowerShell/PSReadLine/releases/tag/v2.2.0-beta4). :tada:
